In [3]:
import pandas as pd
import numpy as np

In [4]:
data = pd.read_csv('E:\paul\AAFT\stct\Salary_Data.csv')


<>:1: SyntaxWarning: invalid escape sequence '\p'
<>:1: SyntaxWarning: invalid escape sequence '\p'
C:\Users\Tinku\AppData\Local\Temp\ipykernel_9808\2639018197.py:1: SyntaxWarning: invalid escape sequence '\p'
  data = pd.read_csv('E:\paul\AAFT\stct\Salary_Data.csv')


In [5]:
data.head()

YearsExperience   Salary
0              1.1  39343.0
1              1.3  46205.0
2              1.5  37731.0
3              2.0  43525.0
4              2.2  39891.0

slope = n  sum(xy) - sum(x)*sum(y)
        ---------------------------
        n sum(x^2)  - sum(x)^2

intercept =  sum(y) -slope* sum(x)
        ----------------------------
                    n 

1. Wht will be the salary of individual having 5 year's of experience

In [ ]:
#slope = n  sum(xy) - sum(x)*sum(y)
#        ---------------------------
#        n sum(x^2)  - sum(x)^2

In [ ]:
#intercept =  sum(y) -slope* sum(x)
#        ----------------------------
#                    n 

In [6]:
a = (data['YearsExperience'] * data['Salary'] ).sum()

a = a* len(data)

print(a)

429658830.0


In [7]:
b = data['Salary'].sum() * data['YearsExperience'].sum()

print(b)

363446346.0


In [8]:
c = (data['YearsExperience']**2).sum()
c = c *len(data)
c

32415.0

In [9]:
d = (data['YearsExperience'].sum())**2
d

25408.36

In [10]:
slope =  (a-b) / (c-d)
slope

9449.962321455077

In [11]:
inter = ( data['Salary'].sum() - slope * data['YearsExperience'].sum()) / len(data)
inter

25792.200198668685

In [12]:
print( 'liner regreesion equation is ' , "y = " ,  round(slope,2) , "x + " ,  round(inter,2) )

liner regreesion equation is  y =  9449.96 x +  25792.2


Here value of X is 5

In [ ]:

x = 5 
y = slope *x + inter
round(y,2)

73042.01

The salary of individual having 5 year's of experience is 73042.01

2. What will be the years of experience an individual having 50,000 salary

slope = n  sum(xy) - sum(x)*sum(y)
        ---------------------------
        n sum(x^2)  - sum(x)^2

In [14]:
#slope = n  sum(xy) - sum(x)*sum(y)
#        ---------------------------
#        n sum(x^2)  - sum(x)^2

In [15]:
g = (data['Salary']**2).sum()
g = g *len(data)
g

5852659743660.0

In [16]:
h = (data['Salary'].sum())**2
h

5198810408100.0

In [17]:
slope =  (a-b) / (g-h)

slope

0.00010126565922605279

In [18]:
data.head()

YearsExperience   Salary
0              1.1  39343.0
1              1.3  46205.0
2              1.5  37731.0
3              2.0  43525.0
4              2.2  39891.0

intercept =  sum(y) -slope* sum(x)
        ----------------------------
                    n 

In [19]:
#intercept =  sum(y) -slope* sum(x)
#        ----------------------------
#                    n 

In [20]:
inte = ( data['YearsExperience'].sum() - slope * data['Salary'].sum()) / len(data)
inte

-2.383160564824357

Here value of Y is 50,000

In [22]:
y = 50000
x = slope * y + inte
round(x,2)

2.68

The years of experience an individual having 50,000 salary is 2.68 years 